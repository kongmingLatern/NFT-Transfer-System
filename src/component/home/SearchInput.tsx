import { Icon } from '@iconify-icon/react'

export default function SearchInput({ search }) {
  return (
    <>
      <input
        type="text"
        placeholder="请输入要查找的物品"
        className="input input-bordered text-black"
        onKeyDown={e => {
          return e.code === 'Enter' ? search(e.target.value) : null
        }}
      />
      <Icon
        icon="ic:outline-search"
        className="absolute right-2 top-[15%] z-1 rounded-lg"
        style={{ color: '#ccc' }}
        width="30"
        height="30"
      />
    </>
  )
}
