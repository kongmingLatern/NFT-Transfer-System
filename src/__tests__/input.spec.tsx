import SearchInput from '@/component/home/SearchInput'
describe('input', () => {
  it('onSearch', () => {
    const jsx = (
      <SearchInput search={() => console.log('search')} />
    )
    expect(jsx.props).toMatchInlineSnapshot(`
      {
        "search": [Function],
      }
    `)
  })
})
