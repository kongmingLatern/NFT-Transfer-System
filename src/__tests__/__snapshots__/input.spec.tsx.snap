// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`input > onSearch 1`] = `
{
  "search": [Function],
}
`;
